sympy/physics/mechanics/_geometry.py

```diff
@@ -0,0 +1,570 @@
+"""Implementations of geometry objects for use by wrapping pathways.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+from sympy.core.backend import Integer, acos, pi, sqrt, sympify, tan
+from sympy.functions.elementary.trigonometric import atan2
+from sympy.polys.polytools import cancel
+from sympy.simplify.simplify import posify, trigsimp
+
+if TYPE_CHECKING:
+    from sympy.core.backend import Symbol
+    from sympy.core.expr import Expr
+    from sympy.physics.mechanics import Point, Vector
+
+
+__all__ = [
+    'Cylinder',
+    'Sphere',
+]
+
+
+class GeometryBase(ABC):
+    """Abstract base class for all geometry classes to inherit from.
+
+    Notes
+    =====
+
+    Instances of this class cannot be directly instantiated by users. However,
+    it can be used to created custom geometry types through subclassing.
+
+    """
+
+    @abstractmethod
+    def _point_is_on_surface(self, point: Point) -> bool:
```

Comment:
These kind of symbolic comparisons are rather complex, as you've already noticed and I see them a lot in this PR. The nice thing about symbolic comparisons is that they are most of the time to sensitive in saying something is not equal, so you'll just be getting an error somewhere. Have you done some research whether these kind of comparisons are also used in other locations within sympy? Would be good to know what is the best method of comparing. To just name a few:
1. `trigsimp(answer) == expected`
2. `trigsimp(answer - expected) == 0`
3. `(answer - expected)._random() == 0` (only works for symbols, dynamicsymbols will return `None`

The problem is that I currently see multiple tricks to catch special examples, which seems to me as suboptimal.

Reply:
Thanks @oscarbenjamin, this was a really helpful explanation.